geode-junit/src/main/java/org/apache/geode/test/junit/rules/Folder.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.test.junit.rules;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.io.FileUtils.deleteDirectory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class Folder {
+
+  private final Path root;
+
+  public Folder(Path root) throws IOException {
+    requireNonNull(root);
+    this.root = Files.createDirectories(root.normalize().toAbsolutePath());
```

Comment:
This should probably be `root.toAbsolutePath().normalize()`.

Reply:
I've fixed this one, but I think I might have this snippet repeated somewhere else in the PR.